### PR TITLE
VS2017 and X64 port

### PR DESCRIPTION
**Pull request by foxesknow**
As well as adding support for Visual Studio 2017 I've also made changes so that the code will build for X64. The majority of the changes are around the socket layer as the code assumed that a socket has represented by an int, but on Windows it's actually a SOCKET. This was causing multiple build warnings as int is 32bit on X64, but SOCKET is 64bit.

### CODE REVIEW

**Comment by danielgbrown** [comment]
Hi,

When running the build_vs17 batch I get the following error.  It is straightforward to fix by opening Visual Studio and re-targeting to the SDK version on my machine but I thought I would let you know.

thanks,

Daniel

f:\program files\Common7\IDE\VC\VCTargets\Platforms\Win32\PlatformToolsets\v141\Toolset.targets(34,5): error MSB8036: The Windows SDK version 10.0.14393.0 was not found. Install the required version of Windows SDK or change the SDK version in the project property pages or by right-clicking the solution and selecting "Retarget solution". [F:\quickfix-master\src\C++\quickfix_vs17.vcxproj]

**Comment by leleftheriades** [comment]
It's a shame that the x64 port was never merged. we really need this

File: src/C++/SocketMonitor.cpp [hunk 1]
```diff
@@ -229,7 +229,7 @@ void SocketMonitor::processReadSet( Strategy& strategy, fd_set& readSet )
 #ifdef _MSC_VER
   for ( unsigned i = 0; i < readSet.fd_count; ++i )
   {
-    int s = readSet.fd_array[ i ];
+    socket_handle s = readSet.fd_array[ i ];
     if( s == m_interrupt )
     {
       int socket = 0;
```
Comment (leleftheriades) [inline src/C++/SocketMonitor.cpp]: This should be socket_handle in order to match your implementation of the signal method. (I assume m_interrupt is the socket that receives the data sent by m_signal)

**Comment by leleftheriades** [comment]
Now that Quickfix uses OpenSSL, unfortunately it assumes sockets to be int, it has the following comment in the code about this:

Even though sizeof(SOCKET) is 8, it's safe to cast it to int, because
the value constitutes an index in per-process table of limited size
and not a real pointer. And we also depend on fact that all processors
Windows run on happen to be two's-complement, which allows to
interchange INVALID_SOCKET and -1.

**Pull request closed without merging**